infra/modules/providers/azure/app-service/main.tf

```diff
@@ -14,25 +17,34 @@ data "azurerm_app_service_plan" "appsvc" {
 }
 
 resource "azurerm_app_service" "appsvc" {
-  name                = "${lower(element(keys(var.app_service_name), count.index))}-${lower(terraform.workspace)}"
+  name                = "${lower(element(keys(var.app_service_config), count.index))}-${lower(terraform.workspace)}"
   resource_group_name = data.azurerm_resource_group.appsvc.name
   location            = data.azurerm_resource_group.appsvc.location
   app_service_plan_id = data.azurerm_app_service_plan.appsvc.id
   tags                = var.resource_tags
-  count               = length(keys(var.app_service_name))
+  count               = length(keys(var.app_service_config))
```

Comment:
if taking the above suggestion this would look like:
`length(locals.app_names)`